google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/Watch.java

```diff
@@ -172,7 +172,7 @@ static Watch forQuery(Query query) {
     target.setQuery(
         QueryTarget.newBuilder()
             .setStructuredQuery(query.buildQuery())
-            .setParent(query.getResourcePath().getParent().getName())
+            .setParent(query.options.getParentPath().getName())
```

Comment:
Any particular reason for direct field access vs. a getter? We provide `getFirestore()` on query, but not `getOptions()`.

Reply:
`options` is purely an internal field that is only used inside the client. We could still use a `getOptions()` here, but I felt that making it package-private was safe enough.